File: INTRODUCCION-POO/EJERCICIO-9/Computadora.py
class Computadora:
    def __init__(self, marca, modelo, procesador, ram_gb, disco_duro_gb):
        self.marca = marca
        self.modelo = modelo
        self.procesador = procesador
        self.ram_gb = ram_gb
        self.disco_duro_gb = disco_duro_gb
        self.encendida = False

    def mostrar_componentes(self):
        print("Componentes principales de la computadora:")
        print(f"  Marca: {self.marca}")
        print(f"  Modelo: {self.modelo}")
        print(f"  Procesador: {self.procesador}")
        print(f"  RAM: {self.ram_gb} GB")
        print(f"  Disco Duro: {self.disco_duro_gb} GB")

    def mostrar_estado(self):
        if self.encendida:
            print("La computadora esta encendida.")
        else:
            print("La computadora esta apagada.")

    def encender(self):
        if not self.encendida:
            self.encendida = True
            print("Encendiendo la computadora...")
        else:
            print("La computadora ya esta encendida.")

    def apagar(self):
        if self.encendida:
            self.encendida = False
            print("Apagando la computadora...")
        else:
            print("La computadora ya esta apagada.")

# Crear una instancia de la computadora
mi_computadora = Computadora("Dell", "Inspiron 15", "Intel Core i5", 8, 512)

# Mostrar los componentes
mi_computadora.mostrar_componentes()
print("-" * 30)

# Mostrar el estado inicial
mi_computadora.mostrar_estado()
print("-" * 30)

# Encender la computadora
mi_computadora.encender()
mi_computadora.mostrar_estado()
print("-" * 30)

# Intentar encenderla de nuevo
mi_computadora.encender()
print("-" * 30)

# Apagar la computadora
mi_computadora.apagar()
mi_computadora.mostrar_estado()
print("-" * 30)

# Intentar apagarla de nuevo
mi_computadora.apagar()

File: INTRODUCCION-POO/EJERCICIO-9/Computadora.java
public class Computadora {
    private String marca;
    private String modelo;
    private String procesador;
    private int ramGb;
    private int discoDuroGb;
    private boolean encendida;

    public Computadora(String marca, String modelo, String procesador, int ramGb, int discoDuroGb) {
        this.marca = marca;
        this.modelo = modelo;
        this.procesador = procesador;
        this.ramGb = ramGb;
        this.discoDuroGb = discoDuroGb;
        this.encendida = false;
    }

    public void mostrarComponentes() {
        System.out.println("Componentes principales de la computadora:");
        System.out.println("  Marca: " + this.marca);
        System.out.println("  Modelo: " + this.modelo);
        System.out.println("  Procesador: " + this.procesador);
        System.out.println("  RAM: " + this.ramGb + " GB");
        System.out.println("  Disco Duro: " + this.discoDuroGb + " GB");
    }

    public void mostrarEstado() {
        if (this.encendida) {
            System.out.println("La computadora esta encendida.");
        } else {
            System.out.println("La computadora esta apagada.");
        }
    }

    public void encender() {
        if (!this.encendida) {
            this.encendida = true;
            System.out.println("Encendiendo la computadora...");
        } else {
            System.out.println("La computadora ya esta encendida.");
        }
    }

    public void apagar() {
        if (this.encendida) {
            this.encendida = false;
            System.out.println("Apagando la computadora...");
        } else {
            System.out.println("La computadora ya esta apagada.");
        }
    }

    public static void main(String[] args) {
        // Crear una instancia de la computadora
        Computadora miComputadora = new Computadora("Dell", "Inspiron 15", "Intel Core i5", 8, 512);

        // Mostrar los componentes
        miComputadora.mostrarComponentes();
        System.out.println("-".repeat(30));

        // Mostrar el estado inicial
        miComputadora.mostrarEstado();
        System.out.println("-".repeat(30));

        // Encender la computadora
        miComputadora.encender();
        miComputadora.mostrarEstado();
        System.out.println("-".repeat(30));

        // Intentar encenderla de nuevo
        miComputadora.encender();
        System.out.println("-".repeat(30));

        // Apagar la computadora
        miComputadora.apagar();
        miComputadora.mostrarEstado();
        System.out.println("-".repeat(30));

        // Intentar apagarla de nuevo
        miComputadora.apagar();
    }
}
